lms/static/styles/components/_Dialog.scss

```diff
@@ -26,10 +21,13 @@
 }
 
 .Dialog__content {
+  display: flex;
+  flex-direction: column;
+  max-height: 90vh;
+  max-width: 700px;
```

Comment:
On master, a narrow screen is responsive, except for any `<pre>` blocks:

<img width="322" alt="Screenshot 2020-06-02 15 27 10" src="https://user-images.githubusercontent.com/2458/83532073-b2a75c00-a4e5-11ea-98d4-8a4d7d9803b4.png">

On this branch, the dialog is not responsive below 700px and requires scrolling in both directions:

<img width="321" alt="Screenshot 2020-06-02 15 27 37" src="https://user-images.githubusercontent.com/2458/83532166-cd79d080-a4e5-11ea-9a9e-7957901c20e6.png">

One way to handle this without media queries would be to use the CSS `min` function:

```diff
diff --git a/lms/static/styles/components/_Dialog.scss b/lms/static/styles/components/_Dialog.scss
index 85b9d2ff..2d84a822 100644
--- a/lms/static/styles/components/_Dialog.scss
+++ b/lms/static/styles/components/_Dialog.scss
@@ -24,7 +24,7 @@
   display: flex;
   flex-direction: column;
   max-height: 90vh;
-  max-width: 700px;
+  max-width: min(700px, 90vw);
   background-color: white;
   border-radius: 3px;
   margin: auto;
```

This requires a pretty modern browser (Chrome 79, Firefox 75, Safari 11) so depending on the environment a fallback may be required for older browsers, eg. `max-width: 90vw` on the line above the `max-width: min(...)`.

Reply:
Due to the amount of space taken up by the Canvas UI, this can be a problem on even moderately narrow desktop screens. I took this crop from a screenshot of a Canvas tab in a 1200px wide window:

<img width="641" alt="Screenshot 2020-06-02 15 38 09" src="https://user-images.githubusercontent.com/2458/83533193-25fd9d80-a4e7-11ea-94d2-01312ebf964c.png">